services/src/main/java/org/keycloak/services/resources/admin/ClientRoleMappingsResource.java

```diff
@@ -173,6 +175,9 @@ public void addClientRoleMapping(List<RoleRepresentation> roles) {
                 auth.roles().requireMapRole(roleModel);
                 user.grantRole(roleModel);
             }
+        } catch (ModelIllegalStateException e) {
+            logger.error(e.getMessage(), e);
+            throw ErrorResponse.error(e.getMessage(), Response.Status.INTERNAL_SERVER_ERROR);
```

Comment:
Created https://github.com/keycloak/keycloak/issues/28587 to improve how we are handling exceptions in our JAX-RS layer.